### ISSUE TITLE
After update to 1.16.1, application crashes on older macOS version 13.0

### ISSUE THREAD

**Issue by ancwrd1** [slint-ui/slint#11499]
### Bug Description

The issue seems to be with the `NSView.displayLink(target:selector:)` API which is only available starting with version 14.0.

As it wasn't mentioned in the CHANGELOG, I think it should be reported as a bug.

Crash dump:

```text

thread 'main' (29327) panicked at /Users/testuser/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/objc2-app-kit-0.3.2/src/generated/NSView.rs:1705:5:
invalid message send to -[WinitView displayLinkWithTarget:selector:]: method not found
stack backtrace:
   0: __rustc::rust_begin_unwind
             at /rustc/e408947bfd200af42db322daf0fadfe7e26d3bd1/library/std/src/panicking.rs:689:5
   1: core::panicking::panic_fmt
             at /rustc/e408947bfd200af42db322daf0fadfe7e26d3bd1/library/core/src/panicking.rs:80:14
   2: objc2::runtime::message_receiver::panic_verify
             at /Users/testuser/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/objc2-0.6.4/src/runtime/message_receiver.rs:348:5
   3: objc2::runtime::message_receiver::msg_send_check_class
             at /Users/testuser/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/objc2-0.6.4/src/runtime/message_receiver.rs:336:5
   4: objc2::runtime::message_receiver::msg_send_check
             at /Users/testuser/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/objc2-0.6.4/src/runtime/message_receiver.rs:308:5
   5: objc2::runtime::message_receiver::MessageReceiver::send_message
             at /Users/testuser/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/objc2-0.6.4/src/runtime/message_receiver.rs:428:13
   6: <MethodFamily as objc2::__macro_helpers::msg_send_retained::MsgSend<Receiver,Return>>::send_message
             at /Users/testuser/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/objc2-0.6.4/src/__macro_helpers/msg_send_retained.rs:35:28
   7: objc2_app_kit::generated::__NSView::NSView::displayLinkWithTarget_selector
             at /Users/testuser/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/objc2-0.6.4/src/macros/extern_methods.rs:238:14
   8: i_slint_backend_winit::frame_throttle::macos_display_link::try_create
             at /Users/testuser/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/i-slint-backend-winit-1.16.1/frame_throttle/macos_display_link.rs:92:41
   9: i_slint_backend_winit::frame_throttle::create_frame_throttle
             at /Users/testuser/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/i-slint-backend-winit-1.16.1/frame_throttle.rs:23:13
  10: i_slint_backend_winit::winitwindowadapter::WinitWindowAdapter::ensure_window
             at /Users/testuser/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/i-slint-backend-winit-1.16.1/winitwindowadapter.rs:501:30
  11: i_slint_backend_winit::SharedBackendData::create_inactive_windows::{{closure}}
             at /Users/testuser/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/i-slint-backend-winit-1.16.1/lib.rs:696:69
  12: core::option::Option<T>::and_then
             at /rustc/e408947bfd200af42db322daf0fadfe7e26d3bd1/library/core/src/option.rs:1546:24
  13: i_slint_backend_winit::SharedBackendData::create_inactive_windows
             at /Users/testuser/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/i-slint-backend-winit-1.16.1/lib.rs:696:54
  14: <i_slint_backend_winit::event_loop::EventLoopState as winit::application::ApplicationHandler<i_slint_backend_winit::SlintEvent>>::about_to_wait
             at /Users/testuser/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/i-slint-backend-winit-1.16.1/event_loop.rs:599:52
  15: winit::event_loop::dispatch_event_for_app
             at /Users/testuser/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/event_loop.rs:647:35
  16: winit::platform::run_on_demand::EventLoopExtRunOnDemand::run_app_on_demand::{{closure}}
             at /Users/testuser/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/platform/run_on_demand.rs:76:13
  17: winit::platform_impl::macos::event_loop::map_user_event::{{closure}}
             at /Users/testuser/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/platform_impl/macos/event_loop.rs:174:22
  18: <alloc::boxed::Box<F,A> as core::ops::function::FnMut<Args>>::call_mut
             at /rustc/e408947bfd200af42db322daf0fadfe7e26d3bd1/library/alloc/src/boxed.rs:2213:9
  19: winit::platform_impl::macos::event_handler::EventHandler::handle_event
             at /Users/testuser/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/platform_impl/macos/event_handler.rs:125:17
  20: winit::platform_impl::macos::app_state::ApplicationDelegate::handle_event
             at /Users/testuser/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/platform_impl/macos/app_state.rs:324:36
  21: winit::platform_impl::macos::app_state::ApplicationDelegate::cleared
             at /Users/testuser/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/platform_impl/macos/app_state.rs:392:14
  22: winit::platform_impl::macos::observer::control_flow_end_handler::{{closure}}
             at /Users/testuser/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/platform_impl/macos/observer.rs:84:80
  23: winit::platform_impl::macos::observer::control_flow_handler::{{closure}}
             at /Users/testuser/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/platform_impl/macos/observer.rs:46:9
```

### Reproducible Code (if applicable)

```slint

```

### Environment Details

- Slint Version: 1.16.1
- Platform/OS: macOS
- Programming Language: Rust
- Backend/Renderer: femtovg

**Comment by ogoffart** [comment]
Thanks for reporting the bug.

Indeed, macOS 13.0 is no longer part of our automated test suite (not available as github action runner) and therefore no longer supported.

That said, the breakage is not intentional and if there is an easy way to restore the compatibility, we will do it.

**Comment by tronical** [comment]
Yes this should be easy to fix :)

**Comment by eira-fransham** [comment]
Just writing this here in case any lurkers are having the same issue, since it's not mentioned in the original description: this isn't present on 1.16.0, so if you're having issues you should be able to downgrade. The main changes between 1.16.0 and 1.16.1 were to improve API semver hygiene, so the only downside to rolling back to 1.16.0 is that you may potentially need to change some code when upgrading to 1.17.0.
